src/drivers/intel/ssp/mn.c

```diff
@@ -65,12 +65,16 @@ static SHARED_DATA struct mn mn;
 
 void mn_init(struct sof *sof)
 {
-#if CONFIG_INTEL_MN
 	int i;
-#endif
 
 	sof->mn = platform_shared_get(&mn, sizeof(mn));
 
+	sof->mn->mclk_source_clock = 0;
+
+	/* initialize the ref counts for mclk */
+	for (i = 0; i < DAI_NUM_SSP_MCLK; i++)
+		sof->mn->mclk_sources_ref[i] = 0;
```

Comment:
Shouldn't this ref count be atomic type ? (since more that one core could be using the clocks)

Reply:
Isn't holding mn->lock enough for multi-cores?

Reply:
it is, but could not see this in the diff.